samples/snippets/src/test/java/com/example/translate/Retry.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.translate;
+
+import java.util.Objects;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class Retry implements TestRule {
+  private int retryCount;
+
+  public Retry(int retryCount) {
+    this.retryCount = retryCount;
+  }
+
+  @Override
+  public Statement apply(Statement base, Description description) {
+    return statement(base, description);
+  }
+
+  private Statement statement(final Statement base, final Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        Throwable caughtThrowable = null;
+
+        // implement retry logic here
+        for (int i = 0; i < retryCount; i++) {
+          try {
+            base.evaluate();
+            return;
+          } catch (Throwable t) {
+            caughtThrowable = t;
+            System.out.println(description.getDisplayName() + ": run " + (i + 1) + " failed");
+          }
```

Comment:
You should have some kind of random wait here so that multiple operations don't hit at the same time.  (also so that the system can recover if there is a problem.  (ideally it will be exponential, starting at 1, 3, 7, etc.)

Reply:
You should do it similarly in the other tests, above.
Start at 1000 * a random value * factor <-- so that multiple tests running at the same time don't do this at the same time. (ie at least one will break free / round).
https://cloud.google.com/storage/docs/exponential-backoff for more info